diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/api.py odoo/api.py
--- odoo.orig/api.py	2019-01-26 05:51:12.000000000 +0000
+++ odoo/api.py	2019-03-14 17:43:55.859900293 +0000
@@ -57,6 +57,10 @@
 
 from odoo.tools import frozendict, classproperty
 
+from functools import partial
+from odoo.tools import config
+import odoo.csv_writer as cw
+
 _logger = logging.getLogger(__name__)
 
 # The following attributes are used, and reflected on wrapping methods:
@@ -664,22 +668,318 @@
     return decorator(func) if not hasattr(func, '_api') else func
 
 
+def get_user_id(obj):
+    if hasattr(obj, "env") and hasattr(obj.env, "uid"):
+        return obj.env.uid
+    return -1
+
+
+def csv_write(user_id, method_name, recs, params, ids = None):
+    # multi have ids, models do not
+    csvw = cw.CsvWriter(config.misc['csv']['location'])
+    recs_name = method_name + '.' + recs._name
+    writer = csv_write.dispatch.get(recs_name, None)
+    if writer != None:
+        writer(csvw, user_id, method_name, recs, params, ids)
+csv_write.dispatch = {}
+
+
+def just_log_it_loudly(csvw, user_id, method_name, recs, params, ids):
+    _logger.debug("LOUDLY! -- User: %s, call method: %s, recs: %s, ids: %s, params: %s", user_id, method_name, recs, ids, params)
+
+
+def create_stage(csvw, user_id, method_name, recs, params, _):
+    csvw.write({
+        'actionName': "Create Stage",
+        'userId': user_id,
+        'id': params.args[0],
+    })
+    _logger.debug("AAAAAAAAAAHHHHHHHHHH!!!!!!!!!")
+csv_write.dispatch['name_create.note.stage'] = create_stage
+
+
+def first_id_action(action, csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': action,
+        'userId': user_id,
+        'id': ids[0],
+    })
+csv_write.dispatch['unlink.note.stage'] = partial(first_id_action, "Delete Stage")
+
+
+def rename_stage(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': "Rename Stage",
+        'userId': user_id,
+        'id': ids[0],
+        'text': params.args[0]['name'],
+    })
+csv_write.dispatch['write.note.stage'] = rename_stage
+
+
+def search(csvw, user_id, method_name, recs, params, _):
+    if not 'domain' in params.kwargs:
+        return
+
+    to_write = {
+        'userId': user_id,
+    }
+    
+    keywords = []
+    tags_or_notes = 0
+
+    for element in params.kwargs['domain']:
+        if not (type(element) == list and len(element) == 3):
+            continue
+        if element[0] == 'memo':
+            tags_or_notes = tags_or_notes | 1
+            keywords.append(element[2])
+        elif element[0] == 'tag_ids':
+            tags_or_notes = tags_or_notes | 2
+            keywords.append(element[2])
+
+    if tags_or_notes == 0:
+        return
+    elif tags_or_notes == 1:
+        to_write['actionName'] = 'Search with Note'
+    elif tags_or_notes == 2:
+        to_write['actionName'] = 'Search with Tag'
+    elif tags_or_notes == 3:
+        to_write['actionName'] = 'Search with both Tag and Note'
+
+    to_write['text'] = ', '.join(keywords)
+    csvw.write(to_write)
+csv_write.dispatch['read_group.note.note'] = search
+
+
+def create_note(csvw, user_id, method_name, recs, params, _):
+    csvw.write({
+        'actionName': 'Create Note',
+        'userId': user_id,
+        'id': params.args[0]['tag_ids'][0][0],
+        'text': params.args[0]['memo'],
+    })
+    open_note.ignored[user_id] = 1
+csv_write.dispatch['create.note.note'] = create_note
+
+
+def create_note_dash(csvw, user_id, method_name, recs, params, _):
+    csvw.write({
+        'actionName': 'Create Note',
+        'userId': user_id,
+        'text': params.args[0],
+    })
+csv_write.dispatch['name_create.note.note'] = create_note_dash
+
+
+csv_write.dispatch['unlink.note.note'] = partial(first_id_action, "Delete Note")
+
+
+def open_note(csvw, user_id, method_name, recs, params, ids):
+    nr_ignore = open_note.ignored.get(user_id, 0)
+    if nr_ignore > 0:
+        open_note.ignored[user_id] = open_note.ignored[user_id] - 1
+        return
+    csvw.write({
+        'actionName': 'Open Note',
+        'userId': user_id,
+        'id': ids[0],
+    })
+open_note.ignored = {}
+csv_write.dispatch['read.note.note'] = open_note
+
+
+def write_note_note(csvw, user_id, method_name, recs, params, ids):
+    open_note.ignored[user_id] = 1
+    try:
+        memo = params.args[0]['memo']
+        csvw.write({
+            'actionName': 'Edit Note',
+            'userId': user_id,
+            'id': ids[0],
+            'text': memo,
+        })
+        return
+    except KeyError:
+        pass
+
+    try:
+        stage_id = params.args[0]['stage_id']
+        if write_note_note.user_moving.get(user_id, None) == stage_id:
+            del write_note_note.user_moving[user_id]
+            actionName = 'Move Note'
+        else:
+            actionName = 'Drag and Drop note'
+            open_note.ignored[user_id] = open_note.ignored[user_id] + 1
+        csvw.write({
+            'actionName': actionName,
+            'userId': user_id,
+            'id': ids[0],
+            'stageId': stage_id,
+        })
+        return
+    except KeyError:
+        pass
+
+    try:
+        _ = params.args[0]['tag_ids']
+        csvw.write({
+            'actionName': "Add tag",
+            'userId': user_id,
+            'id': ids[0]
+        })
+        return
+    except KeyError:
+        pass
+
+    try:
+        _ = params.args[0]['color']
+        csvw.write({
+            'actionName': "Change Note Colour",
+            'userId': user_id,
+            'id': ids[0]
+        })
+        return
+    except KeyError:
+        pass
+write_note_note.user_moving = {}
+csv_write.dispatch['write.note.note'] = write_note_note
+
+
+def move_note_step1(csvw, user_id, method_name, recs, params, ids):
+    if type(ids) == int: write_note_note.user_moving[user_id] = ids
+csv_write.dispatch['name_get.note.stage'] = move_note_step1
+
+
+def open_note_commenting(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': 'Open commenting in note',
+        'userId': user_id,
+        'id': ids[0],
+    })
+csv_write.dispatch['message_get_suggested_recipients.note.note'] = open_note_commenting
+
+
+def comment_in_note(csvw, user_id, method_name, recs, params, ids):
+    to_write = { 'userId': user_id, }
+    if len(params.kwargs['attachment_ids']) == 0:
+        to_write['actionName'] = 'Comment in Note'
+    else:
+        to_write['actionName'] = 'Attach file'
+    to_write['text'] = params.kwargs['body']
+    csvw.write(to_write)
+csv_write.dispatch['message_post.note.note'] = comment_in_note
+
+
+def invite_and_add_follower(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': 'Invite and Add follower',
+        'userId': user_id,
+        'id': params.args[0]['res_id'],
+        'partnerId': params.args[0]['partner_ids'][0][0],
+        'text': params.args[0]['message'],
+    })
+    open_note.ignored[user_id] = 1
+csv_write.dispatch['create.mail.wizard.invite'] = invite_and_add_follower
+
+
+def invite_user(csvw, user_id, method_name, recs, params, _):
+    csvw.write({
+        'actionName': 'Invite User',
+        'userId': user_id,
+        'partnerId': params.args[0],
+    })
+csv_write.dispatch['web_dashboard_create_users.res.users'] = invite_user
+
+
+def create_new_tag(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': 'Create new tag',
+        'userId': user_id,
+        'text': params.args[0]['name'],
+    })
+csv_write.dispatch['create.note.tag'] = create_new_tag
+
+
+def create_new_tag2(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': 'Create new tag',
+        'userId': user_id,
+        'text': params.args[0],
+    })
+csv_write.dispatch['name_create.note.tag'] = create_new_tag2
+
+
+def remove_follower_from_note(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': "Remove follower from note",
+        'userId': user_id,
+        'id': ids[0],
+    })
+    open_note.ignored[user_id] = 1
+csv_write.dispatch['message_unsubscribe.note.note'] = remove_follower_from_note
+
+
+csv_write.dispatch['write.note.tag'] = partial(first_id_action, "Change Tag Colour")
+
+
+def message_post_mail(csvw, user_id, method_name, recs, params, ids):
+    to_write = {
+        'userId': user_id,
+        'text': params.kwargs['body'],
+        'id': ids,
+    }
+
+    user_or_channel = 0
+
+    if len(params.kwargs['partner_ids']) > 0:
+        to_write['partnerId'] = params.kwargs['partner_ids']
+        user_or_channel = user_or_channel | 1
+    if 'class="o_channel_redirect"' in to_write['text']:
+        user_or_channel = user_or_channel | 2
+
+    if user_or_channel == 0:
+        to_write['actionName'] = "Chat with other users"
+    elif user_or_channel == 1:
+        to_write['actionName'] = "Mention user in chat"
+    elif user_or_channel == 2:
+        to_write['actionName'] = "Reference channel in chat"
+    elif user_or_channel == 3:
+        to_write['actionName'] = "Mention user and reference channel in chat"
+    
+    csvw.write(to_write)
+csv_write.dispatch['message_post.mail.channel'] = message_post_mail
+
+
+def remove_mess_channel(csvw, user_id, method_name, recs, params, ids):
+    csvw.write({
+        'actionName': "Remove messaging channel",
+        'userId': user_id,
+    })
+csv_write.dispatch['channel_pin.mail.channel'] = remove_mess_channel
+
+csv_write.dispatch['action_unfollow.mail.channel'] = partial(first_id_action, "Unfollow from a Channel")
+
 
 def call_kw_model(method, self, args, kwargs):
     context, args, kwargs = split_context(method, args, kwargs)
     recs = self.with_context(context or {})
-    _logger.debug("call %s.%s(%s)", recs, method.__name__, Params(args, kwargs))
+    _logger.debug("(Model) ++ User: %s, call method: %s.%s, recs: %s, params: %s", get_user_id(self), method.__name__, recs._name, recs, Params(args, kwargs))
+    csv_write(get_user_id(self), method.__name__, recs, Params(args, kwargs))
     result = method(recs, *args, **kwargs)
     return downgrade(method, result, recs, args, kwargs)
 
+
 def call_kw_multi(method, self, args, kwargs):
     ids, args = args[0], args[1:]
     context, args, kwargs = split_context(method, args, kwargs)
     recs = self.with_context(context or {}).browse(ids)
-    _logger.debug("call %s.%s(%s)", recs, method.__name__, Params(args, kwargs))
+    _logger.debug("(Multi) ++ User: %s, call method: %s.%s, recs: %s, ids: %s, params: %s", get_user_id(self), method.__name__, recs._name, recs, ids, Params(args, kwargs))
+    csv_write(get_user_id(self), method.__name__, recs, Params(args, kwargs), ids)
     result = method(recs, *args, **kwargs)
     return downgrade(method, result, recs, args, kwargs)
 
+
 def call_kw(model, name, args, kwargs):
     """ Invoke the given method ``name`` on the recordset ``model``. """
     method = getattr(type(model), name)
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/csv_writer.py odoo/csv_writer.py
--- odoo.orig/csv_writer.py	1970-01-01 01:00:00.000000000 +0100
+++ odoo/csv_writer.py	2019-03-14 12:02:31.679978434 +0000
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import csv
+from datetime import datetime
+import os
+from os import path
+import threading
+import time
+
+class CsvWriter:
+    writers = {}
+
+    def __init__(self, dirname, column_names_tuple = None):
+        if column_names_tuple == None:
+            self._recover(dirname)
+        else:
+            self._first_init(dirname, column_names_tuple)
+
+    def _recover(self, dirname):
+        dirname = os.path.abspath(dirname)
+        self.__dict__ = CsvWriter.writers[dirname]
+
+    def _first_init(self, dirname, column_names_tuple):
+        dirname = os.path.abspath(dirname)
+        CsvWriter.writers[dirname] = {}
+        self.__dict__ = CsvWriter.writers[dirname]
+
+        self.filename = path.join(os.path.abspath(dirname), time.strftime("%Y%m%d-%H%M%S", time.gmtime()) + '.csv')
+
+        self.write_lock = threading.Lock()
+        self.column_names = {}
+        self.id_to_ip = {}
+
+        label_row = []
+        for idx, column_name in enumerate(column_names_tuple):
+            self.column_names[column_name] = idx
+            label_row.append(column_name)
+
+        with self.write_lock, open(self.filename, 'w') as f:
+            csv.writer(f).writerow(label_row)
+
+    def connect_id_to_ip(self, userId, userIp):
+        with self.write_lock:
+            self.id_to_ip[userId] = userIp
+
+    def write(self, elements_dict):
+        with self.write_lock, open(self.filename, 'a') as f:
+            if 'userId' in elements_dict and not 'ipAddr' in elements_dict:
+                elements_dict['ipAddr'] = self.id_to_ip.get(elements_dict['userId'], '')
+            if not 'timestamp' in elements_dict:
+                elements_dict['timestamp'] = datetime.now()
+
+            row = ['']*len(self.column_names)
+
+            for column_name, value in elements_dict.items():
+                row[ self.column_names[column_name] ] = str(value)
+
+            csv.writer(f).writerow(row)
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/http.py odoo/http.py
--- odoo.orig/http.py	2018-11-23 05:55:29.000000000 +0000
+++ odoo/http.py	2019-03-14 12:02:31.679978434 +0000
@@ -51,6 +51,9 @@
 
 from .modules.module import module_manifest
 
+from odoo.tools import config
+import odoo.csv_writer as cw
+
 _logger = logging.getLogger(__name__)
 rpc_request = logging.getLogger(__name__ + '.rpc.request')
 rpc_response = logging.getLogger(__name__ + '.rpc.response')
@@ -310,6 +313,10 @@
         raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
 
     def _call_function(self, *args, **kwargs):
+        _logger.debug("User: %s Remote addr: %s", self.uid, self.httprequest.remote_addr)
+        csvw = cw.CsvWriter (config.misc['csv']['location'])
+        csvw.connect_id_to_ip(self.uid, self.httprequest.remote_addr)
+
         request = self
         if self.endpoint.routing['type'] != self._request_type:
             msg = "%s, %s: Function declared as capable of handling request of type '%s' but called with a request of type '%s'"
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/odoo.conf odoo/odoo.conf
--- odoo.orig/odoo.conf	1970-01-01 01:00:00.000000000 +0100
+++ odoo/odoo.conf	2019-03-14 12:02:31.679978434 +0000
@@ -0,0 +1,39 @@
+[options]
+addons_path = /mnt/extra-addons
+data_dir = /var/lib/odoo
+; admin_passwd = admin
+; csv_internal_sep = ,
+; db_maxconn = 64
+; db_name = False
+; db_template = template1
+; dbfilter = .*
+; debug_mode = False
+; email_from = False
+; limit_memory_hard = 2684354560
+; limit_memory_soft = 2147483648
+; limit_request = 8192
+; limit_time_cpu = 60
+; limit_time_real = 120
+; list_db = True
+; log_db = False
+log_handler = [':DEBUG']
+log_level = debug
+; logfile = None
+; longpolling_port = 8072
+; max_cron_threads = 2
+; osv_memory_age_limit = 1.0
+; osv_memory_count_limit = False
+; smtp_password = False
+; smtp_port = 25
+; smtp_server = localhost
+; smtp_ssl = False
+; smtp_user = False
+; workers = 0
+; xmlrpc = True
+; xmlrpc_interface = 
+; xmlrpc_port = 8069
+; xmlrpcs = True
+; xmlrpcs_interface = 
+; xmlrpcs_port = 8071
+[csv]
+location = /usr/lib/python3/dist-packages/odoo/csvfolder
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/odoo.conf.orig odoo/odoo.conf.orig
--- odoo.orig/odoo.conf.orig	1970-01-01 01:00:00.000000000 +0100
+++ odoo/odoo.conf.orig	2019-03-14 12:02:31.679978434 +0000
@@ -0,0 +1,39 @@
+[options]
+addons_path = /mnt/extra-addons
+data_dir = /var/lib/odoo
+; admin_passwd = admin
+; csv_internal_sep = ,
+; db_maxconn = 64
+; db_name = False
+; db_template = template1
+; dbfilter = .*
+; debug_mode = False
+; email_from = False
+; limit_memory_hard = 2684354560
+; limit_memory_soft = 2147483648
+; limit_request = 8192
+; limit_time_cpu = 60
+; limit_time_real = 120
+; list_db = True
+; log_db = False
+log_handler = [':DEBUG']
+log_level = debug
+; logfile = None
+; longpolling_port = 8072
+; max_cron_threads = 2
+; osv_memory_age_limit = 1.0
+; osv_memory_count_limit = False
+; smtp_password = False
+; smtp_port = 25
+; smtp_server = localhost
+; smtp_ssl = False
+; smtp_user = False
+; workers = 0
+; xmlrpc = True
+; xmlrpc_interface = 
+; xmlrpc_port = 8069
+; xmlrpcs = True
+; xmlrpcs_interface = 
+; xmlrpcs_port = 8071
+[csv]
+location = /usr/lib/python3/dist-packages/odoo/csvfolder
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/odoo.conf.rej odoo/odoo.conf.rej
--- odoo.orig/odoo.conf.rej	1970-01-01 01:00:00.000000000 +0100
+++ odoo/odoo.conf.rej	2019-03-14 12:02:31.679978434 +0000
@@ -0,0 +1,11 @@
+--- odoo.orig/odoo.conf	2019-03-11 17:09:45.000000000 +0000
++++ odoo/odoo.conf	2019-03-13 17:18:12.397776321 +0000
+@@ -35,5 +35,6 @@
+ ; xmlrpcs = True
+ ; xmlrpcs_interface = 
+ ; xmlrpcs_port = 8071
+-[cvs]
+-location = /usr/lib/python3/dist-packages/odoo/analyser_log.cvs
++[csv]
++location = /usr/lib/python3/dist-packages/odoo/csvfolder
++
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/service/server.py odoo/service/server.py
--- odoo.orig/service/server.py	2018-11-03 05:47:26.000000000 +0000
+++ odoo/service/server.py	2019-03-14 12:02:31.679978434 +0000
@@ -227,7 +227,6 @@
         while True:
             time.sleep(SLEEP_INTERVAL + number)     # Steve Reich timing style
             registries = odoo.modules.registry.Registry.registries
-            _logger.debug('cron%d polling for jobs', number)
             for db_name, registry in registries.items():
                 if registry.ready:
                     try:
@@ -823,7 +822,6 @@
     def process_work(self):
         rpc_request = logging.getLogger('odoo.netsvc.rpc.request')
         rpc_request_flag = rpc_request.isEnabledFor(logging.DEBUG)
-        _logger.debug("WorkerCron (%s) polling for jobs", self.pid)
         db_names = self._db_list()
         if len(db_names):
             self.db_index = (self.db_index + 1) % len(db_names)
diff -x '*.pyc' -x '*.csv' -x csvfolder -Nru odoo.orig/tools/config.py odoo/tools/config.py
--- odoo.orig/tools/config.py	2018-10-12 05:54:23.000000000 +0100
+++ odoo/tools/config.py	2019-03-14 12:02:31.679978434 +0000
@@ -12,6 +12,7 @@
 import sys
 import odoo
 from .. import release, conf, loglevels
+from .. import csv_writer as cw
 from . import appdirs, pycompat
 
 from passlib.context import CryptContext
@@ -550,6 +551,16 @@
             pass
         except ConfigParser.NoSectionError:
             pass
+        cw.CsvWriter(self.misc['csv']['location'], (
+            'timestamp',
+            'actionName',
+            'userId',
+            'ipAddr',
+            'id',
+            'stageId',
+            'partnerId',
+            'text',
+        ))
 
     def save(self):
         p = ConfigParser.RawConfigParser()
